File: python3/Path Sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if not root:
            return False

        # Check if the current node is a leaf node and its value equals the remaining target sum
        if not root.left and not root.right and root.val == targetSum:
            return True

        # Recur for the left and right subtrees
        return self.hasPathSum(root.left, targetSum - root.val) or self.hasPathSum(root.right, targetSum - root.val)

File: bin/go/Path Sum.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func hasPathSum(root *TreeNode, targetSum int) bool {
    if root == nil {
        return false
    }

    // Check if the current node is a leaf node and its value equals the remaining target sum
    if root.Left == nil && root.Right == nil && root.Val == targetSum {
        return true
    }

    // Recur for the left and right subtrees
    return hasPathSum(root.Left, targetSum-root.Val) || hasPathSum(root.Right, targetSum-root.Val)
}